stock_picking_type_bypass_reservation/models/stock_move.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2025 Moduon Team S.L.
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl-3.0)
+
+from odoo import fields, models
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    def _should_bypass_reservation(self, forced_location=False):
+        res = super()._should_bypass_reservation(forced_location=forced_location)
+        return res or self.env.context.get("force_bypass_reservation")
+
+    def _action_assign(self, force_qty=False):
+        # Do a first run to gather items in stock and then bypass the reservation for
+        # the rest. FIXME: This won't work for items without tracking as the first
+        # reserved line will be set to update with the whole move demand.
+        res = super()._action_assign(force_qty=force_qty)
+        moves_to_test_bypass = self.filtered(
+            lambda x: x.picking_type_id
+            and x.picking_type_id.should_bypass_reservation()
+        )
+        if moves_to_test_bypass:
+            moves_to_test_bypass.with_context(force_bypass_reservation=True)
+            super(
+                StockMove,
+                moves_to_test_bypass.with_context(force_bypass_reservation=True),
+            )._action_assign(force_qty=force_qty)
+        return res
+
+    def _prepare_move_line_vals(self, quantity=None, reserved_quant=None):
+        # Let's flag the bypassed move lines. It'll be handy to avoid issues when
+        # unreserving.
+        vals = super()._prepare_move_line_vals(quantity, reserved_quant)
+        if self.env.context.get("force_bypass_reservation"):
+            vals.update({"bypassed_reservation": True})
+        return vals
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    bypassed_reservation = fields.Boolean()
+
+    def _action_done(self):
+        # Allow to complete without reservation issues. Thes flagged move lines didn't
+        # take their reserve from any quant, and they shouldn't unreserve from any
+        # quant either.
+        bypassed_mls = self.filtered("bypassed_reservation")
+        super(
+            StockMoveLine, bypassed_mls.with_context(force_bypass_reservation=True)
+        )._action_done()
+        return super(StockMoveLine, self - bypassed_mls)._action_done()
+
+    def write(self, vals):
+        # Allow to update without reservation issues. Thes flagged move lines didn't
+        # take their reserve from any quant, and they shouldn't unreserve from any
+        # quant either.
+        bypassed_mls = self.filtered("bypassed_reservation")
+        super(
+            StockMoveLine, bypassed_mls.with_context(force_bypass_reservation=True)
+        ).write(vals)
+        return super(StockMoveLine, self - bypassed_mls).write(vals)
+
+    def unlink(self):
+        # Allow to unlink without reservation issues. Thes flagged move lines didn't
```

Comment:
```suggestion
        # Allow to unlink without reservation issues. These flagged move lines didn't
```

suggestion: consider adding a [module-level docstring](https://stackoverflow.com/a/2557196/1468388) to explain common stuff, instead of repeating the same comment many times.

Reply:
Is that going to work with Odoo's special inheritance?

Reply:
Yes, it's just a docstring at the module level. Odoo does nothing special with that AFAIK.